File: C#/PalindromeNumber.cs
// Given an integer x, return true if x is a , and false otherwise.

public class Solution {
    public bool IsPalindrome(int x) {
        if (x < 0) {
            return false;
        }

        int tmp = x;
        int place = 0;
        while (tmp > 0) {
            ++place;
            tmp /= 10;
        }
        place = (int) Math.Pow(10, place-1);

        int copyX = x;
        int pal = 0;
        while (x > 0) {
            pal += (x%10)*place;
            place /= 10;
            x /= 10;
        }

        return pal==copyX;
    }
}

File: PHP/PalindromeNumber.php
<!-- Given an integer x, return true if x is a , and false otherwise. -->

class Solution {
    /**
     * @param Integer $x
     * @return Boolean
     */
    function isPalindrome($x) {
        if ($x < 0) {
            return false;
        }

        if ($x < 10) {
            return true;
        }

        $cp = $x;
        $arr = [];
        $backwards = 0;
        $place = 1;

        while ($cp > 0) {
            $arr[] = ($cp % 10);
            $cp = floor($cp / 10);
            if ($cp == 0) {
                break;
            }
        }

        for ($i = count($arr)-1; $i > -1; $i--) {
            $backwards += ($place * $arr[$i]);
            $place *= 10;
        }

        return $x == $backwards;
    }
}